src/AWS.Deploy.CLI/ServerMode/Controllers/DeploymentController.cs

```diff
@@ -81,7 +81,19 @@ await _projectParserUtility.Parse(input.ProjectPath)
 
             _stateServer.Save(output.SessionId, state);
 
-            output.DefaultDeploymentName = _cloudApplicationNameGenerator.GenerateValidName(state.ProjectDefinition, new List<CloudApplication>());
+            var deployedApplicationQueryer = serviceProvider.GetRequiredService<IDeployedApplicationQueryer>();
+            var session = CreateOrchestratorSession(state);
+            var orchestrator = CreateOrchestrator(state);
+
+            // Determine what recommendations are possible for the project.
+            var recommendations = await orchestrator.GenerateDeploymentRecommendations();
```

Comment:
Should we save `recommendations` to `state.NewRecommendations` and `existingApplications` to `state.ExistingDeployments` so we don't have to recomputed those? That would mean those controller getter methods should look to see if the state already has a value before doing all the work again.

Reply:
Fixed